### PR TITLE
Issue/redux

### PR DESCRIPTION
**Pull request by BelohvostovAlex**

### CODE REVIEW

File: src/hooks/useFetchData.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const fetchData = async () => {
+      try {
+        setIsLoading(true);
+        const response = await axios.get(url);
+        if (response.status === 200) {
+          setData(response.data);
+          setIsLoading(false);
+          setHasError(false);
+          setErrorMessage('');
+          console.log(response);
```
Comment (igorTT) [inline src/hooks/useFetchData.tsx]: Delete console.log

File: src/hooks/useFetchData.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+
+export const useFetchData = (url: string) => {
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  useEffect(() => {
```
Comment (igorTT) [inline src/hooks/useFetchData.tsx]: Add an empty line

File: src/hooks/useActions.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from '../store';
+import { bindActionCreators } from 'redux';
+
+import { allActionCreators } from '../store/reducers/action-creators';
+
+export const useActions = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  return bindActionCreators(allActionCreators, dispatch);
+};
```
Comment (igorTT) [inline src/hooks/useActions.tsx:10]: 👍

File: src/services/WebService.tsx [hunk 4]
```diff
@@ -21,8 +21,24 @@ class WebService {
...
+          status?: string | undefined;
+          title: string;
+          description: string;
+        }
+      | {
+          status: string;
+          id?: string | undefined;
+          title?: string | undefined;
+          description?: string | undefined;
+        }
```
Comment (igorTT) [inline src/services/WebService.tsx]: I don't understand- what is this

File: src/hooks/useHandlers.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    editTodo: async (todo: IPureTodo) => {
+      const { id } = initialValue;
+      const currTodo = todos.find((todo) => todo.id === id);
+      const editedTodo = {
+        ...currTodo!,
+        ...todo,
+      };
+      updateTodoActionCreator(editedTodo);
+      setVisible(false);
+    },
```
Comment (igorTT) [inline src/hooks/useHandlers.tsx]: I don't see await's

File: src/hooks/useHandlers.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
```
Comment (igorTT) [inline src/hooks/useHandlers.tsx:53]: I see errors
image

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/hooks/useFetchData.tsx (hunk 1 above)
Comment (BelohvostovAlex) [reply src/hooks/useFetchData.tsx]: deleted, changed

File: src/hooks/useFetchData.tsx (hunk 2 above)
Comment (BelohvostovAlex) [reply src/hooks/useFetchData.tsx]: added

File: src/hooks/useHandlers.tsx (hunk 5 above)
Comment (BelohvostovAlex) [reply src/hooks/useHandlers.tsx]: deleted

File: src/hooks/useHandlers.tsx (hunk 6 above)
Comment (BelohvostovAlex) [reply src/hooks/useHandlers.tsx:53]: Sometimes after changing branch, i have the same, but after saving the file manually once again, it works... weird situation, dunno why it happens..

File: src/services/WebService.tsx (hunk 4 above)
Comment (BelohvostovAlex) [reply src/services/WebService.tsx]: me2, deleted

File: src/store/reducers/todosReducer/action-creators.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  SetErrorActionTodos,
+  DeleteTodoActionTodos,
+  UpdateTodoActionTodos,
+  UpdateTodoStatusActionTodos,
+  SetFilterActionTodos,
+} from './types';
+
+const webService = new WebService()
+
+export const TodosActionCreators = {
```
Comment (igorTT) [inline src/store/reducers/todosReducer/action-creators.ts]: Use uppercase naming if only it's a class or typescript entity.
Replace this object with a simple functions. Then export them 
```javascript
export {
  setTodos,
  fetchTodos,
  ...etc
}
```
You can import them as a default value
```javascript
import todosAction from './some_adress';
```

**Review by igorTT — CHANGES_REQUESTED** [review]
Small fixes

File: src/store/reducers/todosReducer/action-creators.ts (hunk 7 above)
Comment (BelohvostovAlex) [reply src/store/reducers/todosReducer/action-creators.ts]: done, noted

**Review by igorTT — APPROVED** [review]

**Pull request merged**